ospfd/ospf_vty.c

```diff
@@ -8860,6 +8862,7 @@ DEFUN (no_ip_ospf_area,
 	if (ospf) {
 		ospf_interface_area_unset(ospf, ifp);
 		ospf->if_ospf_cli_count--;
+		ospf_area_check_free(ospf, area_id);
```

Comment:
Have you tested the scenario? I think this fix won't work because OSPF interface down event gets triggered as an event so the interface will remain present when this function ospf_area_check_free is called hence it won't be able to delete the area.

Reply:
@mobash-rasool you're right if the interface is disabled, the area will not be removed. However I'm not sure if in that situation the area needs to be removed. Anyway this PR was only meant to fix the removal of the area

Reply:
Interface down event is another case where area must not be removed i.e. interface goes down due to link failure and then up after link recovery. In this case, area must be already configured to get again the OSPF adjacency.

So, the proposed PR sounds good if it solve only the removal of `ip ospf xxxx` CLI and not treat the interface down event.